src/app/src/util/embeddedMap.js

```diff
@@ -0,0 +1,100 @@
+import sortBy from 'lodash/sortBy';
+
+import { OARFont } from './constants';
+
+const formatExistingWidth = ({ width = '1000' }) => {
+    const fullWidth = width[width.length - 1] === '%';
+    if (fullWidth) {
+        return { fullWidth, width: '100' };
+    }
+    if (!width.length) {
+        return { fullWidth, width: '1000' };
+    }
+    return { fullWidth, width };
+};
+
+export const formatExistingConfig = embedConfig => {
+    if (!embedConfig) {
+        return {
+            width: '1000',
+            color: '#3d2f8c',
+            font: OARFont,
+            includeOtherContributorFields: false,
+            height: '800',
+        };
+    }
+    return {
+        ...embedConfig,
+        ...formatExistingWidth(embedConfig),
+        color: embedConfig.color.length ? embedConfig.color : '#3d2f8c',
+        font: embedConfig.font.length ? embedConfig.font : OARFont,
+        includeOtherContributorFields:
+            embedConfig.show_other_contributor_information,
+        height: embedConfig.height.length ? embedConfig.height : '800',
+    };
+};
+
+export const formatExistingFields = (fields = []) =>
+    fields.map(field => ({
+        columnName: field.column_name,
+        displayName: field.display_name,
+        visible: field.visible,
+        order: field.order,
+    }));
+
+const doesExist = (field, existingFields) =>
+    existingFields.some(f => f.columnName === field);
+
+const filterAndFormatNonstandardFields = ({
+    nonstandardFields,
+    existingFields,
+}) =>
+    nonstandardFields
+        .filter(field => !doesExist(field, existingFields))
+        .map(field => ({
+            columnName: field,
+            displayName: field,
+            visible: true,
+        }));
+
+export const combineEmbedAndNonstandardFields = (
+    embedFields,
+    nonstandardFields,
+) => {
+    const existingFields = formatExistingFields(embedFields);
+    const newFields = filterAndFormatNonstandardFields({
+        nonstandardFields,
+        existingFields,
+    });
+    return sortBy([...existingFields, ...newFields], f => f.order).map(
+        (f, i) => ({
+            ...f,
+            order: f.order || i,
+        }),
+    );
+};
+
+const formatEmbedFieldsForServer = fields =>
+    sortBy(fields, f => f.order).map((f, i) => ({
+        visible: f.visible,
+        order: i,
+        display_name: f.displayName,
+        column_name: f.columnName,
+    }));
+
+const formatWidthForServer = ({ width, fullWidth }) => {
```

Comment:
When I check the 100% box I am seeing the width field being set to 100.

![2021-05-11 14 47 21](https://user-images.githubusercontent.com/17363/117889134-dfbccc80-b267-11eb-982b-2f2a3326eb19.gif)

Reply:
Updated to make the behavior a little bit more logical in 9d913b8, but I did want to note as well that full-width embed isn't set up for the preview or embed code yet (#1311 will add that), so the required/expected behavior here will likely undergo additional changes when that is implemented.